index.js

```diff
@@ -1,78 +1,96 @@
 
-module.exports = function (app, urls) {
+function SignalHub (app, urls) {
+  if (!(this instanceof SignalHub)) return new SignalHub(app, urls)
+
+  events.EventEmitter.call(this)
+  
   if (!app) throw new Error('app name required')
   if (!urls || !urls.length) throw new Error('signalhub url(s) required')
 
-  var that = {}
   if (!Array.isArray(urls)) urls = [urls]
 
-  urls = urls.map(function (url) {
+  this.app = app
+  this.urls = urls.map(function (url) {
     url = url.replace(/\/$/, '')
     return url.indexOf('://') === -1 ? 'http://' + url : url
   })
+}
 
-  that.subscribe = function (channel) {
-    var endpoint
-
-    if (Array.isArray(channel)) {
-      endpoint = channel.join(',')
-    } else {
-      endpoint = channel
-    }
-
-    var all = through.obj()
-    var streams = urls.map(function (url) {
-      return ess(url + '/v1/' + app + '/' + endpoint, {json: true})
-    })
+inherits(SignalHub, events.EventEmitter)
 
-    if (streams.length === 1) return streams[0]
+SignalHub.prototype.subscribe = function (channel) {
+  var endpoint
+  if (Array.isArray(channel)) {
+    endpoint = channel.join(',')
+  } else {
+    endpoint = channel
+  }
 
-    var all = through.obj()
+  var streams = this.urls.map(function (url) {
+    return ess(url + '/v1/' + this.app + '/' + endpoint, {json: true})
+  })
 
+  var all
+  if (streams.length === 1) {
+    all = streams[0]
+  } else {
+    all = through.obj()
     all.setMaxListeners(0)
     streams.forEach(function (stream) {
       stream.on('open', function () {
         all.emit('open')
       })
       pump(stream, all)
     })
-
-    return all
   }
 
-  var broadcast = function (url, channel, message, cb) {
-    nets({
-      method: 'POST',
-      json: message,
-      url: url + '/v1/' + app + '/' + channel
-    }, function (err, res) {
-      if (err) return cb(err)
-      if (res.statusCode !== 200) return cb(new Error('Bad status: ' + res.statusCode))
-      cb()
-    })
-  }
+  this.on('close', function (cb) {
+    all.destroy()
+  })
 
-  that.broadcast = function (channel, message, cb) {
-    if (!cb) cb = noop
+  return all
+}
 
-    var pending = urls.length
-    var errors = 0
+SignalHub.prototype.broadcast = function (channel, message, cb) {
+  if (!cb) cb = noop
 
-    urls.forEach(function (url) {
-      broadcast(url, channel, message, function (err) {
-        if (err) errors++
-        if (--pending) return
-        if (errors === urls.length) return cb(err)
-        cb()
-      })
+  var pending = this.urls.length
+  var errors = 0
+
+  var self = this
+  return this.urls.forEach(function (url) {
+    var bc = broadcast(url, channel, message, function (err) {
+      if (err) errors++
+      if (--pending) return
+      if (errors === self.urls.length) return cb(err)
+      cb()
     })
-  }
+  })
+}
 
-  return that
+SignalHub.prototype.close = SignalHub.prototype.destroy = function (cb) {
+  this.emit('close')
+  if (typeof cb === 'function') cb()
```

Comment:
technically it would be better to wait for all subscribers to close before emitting close but i'm ok with this for now.